pandas/core/indexes/base.py

```diff
@@ -709,6 +699,170 @@ def dtype_str(self):
         """
         return str(self.dtype)
 
+    def ravel(self, order='C'):
+        """
+        Return an ndarray of the flattened values of the underlying data.
+
+        See Also
+        --------
+        numpy.ndarray.ravel
+        """
+        return self._ndarray_values.ravel(order=order)
+
+    def view(self, cls=None):
+
+        # we need to see if we are subclassing an
+        # index type here
+        if cls is not None and not hasattr(cls, '_typ'):
+            result = self._data.view(cls)
+        else:
+            result = self._shallow_copy()
+        if isinstance(result, Index):
+            result._id = self._id
+        return result
+
+    _index_shared_docs['astype'] = """
+        Create an Index with values cast to dtypes. The class of a new Index
+        is determined by dtype. When conversion is impossible, a ValueError
+        exception is raised.
+
+        Parameters
+        ----------
+        dtype : numpy dtype or pandas type
+        copy : bool, default True
+            By default, astype always returns a newly allocated object.
+            If copy is set to False and internal requirements on dtype are
+            satisfied, the original data is used to create a new Index
+            or the original Index is returned.
+
+            .. versionadded:: 0.19.0
+        """
+
+    @Appender(_index_shared_docs['astype'])
```

Comment:
maybe this goes in conversion

Reply:
I considered that, thought that ndarray-like methods should take priority